pennylane/devices/default_qubit_jax.py

```diff
@@ -214,9 +216,14 @@ def _evolve_state_vector_under_parametrized_evolution(
 
         state = self._flatten(state)
 
+        with jax.ensure_compile_time_eval():
+            H_jax = ParametrizedHamiltonianPytree.from_hamiltonian(
+                operation.H, dense=len(operation.wires) > 4, wire_order=self.wires
+            )
+
         def fun(y, t):
             """dy/dt = -i H(t) y"""
-            return -1j * qml.matrix(operation.H(operation.data, t=t), wire_order=self.wires) @ y
+            return -1j * (H_jax(operation.data, t=t) @ y)
```

Comment:
from a point of view of computational complexity, it would be cheaper to remove the parenthesis and do
```suggestion
            return -1j * H_jax(operation.data, t=t) @ y
```
because this is equivalent to ` (-1j * H_jax(operation.data, t=t)) @ y` which has `N_terms` scalar multiplications instead of `2^N_wires`.

but we need to implement `__rmul_` in `LazyJaxSum`

Reply:
Changed! 🚀 👍